x-pack/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/logic/rule_asset/rule_asset_saved_objects_client.ts

```diff
@@ -27,33 +26,53 @@ export interface IRuleAssetSavedObject {
   attributes: IRuleAssetSOAttributes;
 }
 
-export interface RuleAssetSavedObjectsClient {
-  find: (
-    options?: Omit<SavedObjectsFindOptions, 'type'>
-  ) => Promise<SavedObjectsFindResponse<IRuleAssetSavedObject>>;
-  all: () => Promise<IRuleAssetSavedObject[]>;
+export interface RuleAssetsClient {
```

Comment:
Nit: Let's rename it to `IRuleAssetsClient`. The idea is to start an interface with `I` for objects with behavior, and have interfaces without `I` for data. This allows using classes that implement interfaces if needed (yes I remember you don't like classes!).

```ts
export class RuleAssetsClient implements IRuleAssetsClient, IAnotherInterface { ... }
```

Otherwise, it would be Java 😂 

```ts
export class RuleAssetsClientImpl implements RuleAssetsClient, AnotherInterface { ... }
```